contracts/utils/math/Math.sol

```diff
@@ -17,14 +17,36 @@ library Math {
         Expand // Away from zero
     }
 
+    /**
+     * @dev Unsigned saturating addition, bounds to `2 ** 256 - 1` instead of overflowing.
+     */
+    function saturatingAdd(uint256 a, uint256 b) internal pure returns (uint256) {
+        unchecked {
+            a = a + b;
+            b = 0 - SafeCast.toUint(a < b);
+            return a | b;
+        }
+    }
+
     /**
      * @dev Returns the addition of two unsigned integers, with an success flag (no overflow).
      */
     function tryAdd(uint256 a, uint256 b) internal pure returns (bool success, uint256 result) {
         unchecked {
             uint256 c = a + b;
-            if (c < a) return (false, 0);
-            return (true, c);
+            success = c >= a;
+            // equivalent to: c >= a ? c : 0
+            result = SafeCast.toUint(success) * c;
+        }
+    }
+
+    /**
+     * @dev Unsigned saturating subtraction, bounds to zero instead of overflowing.
+     */
+    function saturatingSub(uint256 a, uint256 b) internal pure returns (uint256) {
+        unchecked {
+            // equivalent to: a >= b ? a - b : 0
+            return (a - b) * SafeCast.toUint(a > b);
```

Comment:
We could also do.
```suggestion
            return ternary(a > b, a - b, 0);
```

Reply:
Otherwise I can refactor it for use the `ternary` at a little gas cost, but still better than branching.